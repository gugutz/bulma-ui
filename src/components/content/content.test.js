/* eslint-disable react/jsx-child-element-spacing */
import React from 'react'
import renderer from 'react-test-renderer'
import Content from './content'

describe('<Content /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Content>
          <h1>Medium Content</h1>
          <p>
            Lorem ipsum
            <sup>
              <a>[1]</a>
            </sup>
            dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus
            ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum
            justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio,
            sollicitudin vel erat vel, interdum mattis neque. Sub
            <sub>script</sub>
            works as well!
          </p>
        </Content>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Content size={size}>
            <h1>Medium Content</h1>
            <p>
              Lorem ipsum
              <sup>
                <a>[1]</a>
              </sup>
              dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus
              ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum
              justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio,
              sollicitudin vel erat vel, interdum mattis neque. Sub
              <sub>script</sub>
              works as well!
            </p>
          </Content>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
