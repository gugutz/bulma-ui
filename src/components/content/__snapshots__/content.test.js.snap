// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> component should render correctly 1`] = `
<div
  className="content"
>
  <h1>
    Medium Content
  </h1>
  <p>
    Lorem ipsum
    <sup>
      <a>
        [1]
      </a>
    </sup>
    dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque. Sub
    <sub>
      script
    </sub>
    works as well!
  </p>
</div>
`;

exports[`<Content /> component should support size prop with large 1`] = `
<div
  className="content is-large"
>
  <h1>
    Medium Content
  </h1>
  <p>
    Lorem ipsum
    <sup>
      <a>
        [1]
      </a>
    </sup>
    dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque. Sub
    <sub>
      script
    </sub>
    works as well!
  </p>
</div>
`;

exports[`<Content /> component should support size prop with medium 1`] = `
<div
  className="content is-medium"
>
  <h1>
    Medium Content
  </h1>
  <p>
    Lorem ipsum
    <sup>
      <a>
        [1]
      </a>
    </sup>
    dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque. Sub
    <sub>
      script
    </sub>
    works as well!
  </p>
</div>
`;

exports[`<Content /> component should support size prop with normal 1`] = `
<div
  className="content"
>
  <h1>
    Medium Content
  </h1>
  <p>
    Lorem ipsum
    <sup>
      <a>
        [1]
      </a>
    </sup>
    dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque. Sub
    <sub>
      script
    </sub>
    works as well!
  </p>
</div>
`;

exports[`<Content /> component should support size prop with small 1`] = `
<div
  className="content is-small"
>
  <h1>
    Medium Content
  </h1>
  <p>
    Lorem ipsum
    <sup>
      <a>
        [1]
      </a>
    </sup>
    dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque. Sub
    <sub>
      script
    </sub>
    works as well!
  </p>
</div>
`;
